### ISSUE TITLE
[BUG] Decoration controller doesn't allow upgrading structure

### ISSUE THREAD

**Issue by dupontct** [ldtteam/minecolonies#8663]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19

### MineColonies Version

1.19.2-1.0.1039

### Structurize Version

1.19.2-1.0.446

### Related Mods and their Versions

1. Forge version: 43.1.25
2. BlockUI version: 1.19-0.0.61
3. Domum Ornamentum version: 1.19-1.0.59

### Current Behavior

I have the blueprints named gatelarge1, gatelarge2, gatelarge3, gatelarge4, and gatelarge5 created using the scan tool.

Each one has decoration controller, but when I build it in world using a builder. The decoration controller doesn't give me the option to upgrade.

The build tool displays them as multiple levels of a single build option.
![Build Tool UI](https://user-images.githubusercontent.com/29468/191386117-682c94e0-aa28-44af-affd-07f1450dc963.png)

But when I right-click on the Decoration Controller there are no buttons at all.
![Decoration Controller UI](https://user-images.githubusercontent.com/29468/191386405-83593ac0-be6b-4f77-bb65-333fe3872487.png)

The log reports that it can't find the gatelarge1 blueprint file without the ".blueprint" behind it.
See attached log

### Expected Behavior

That the Decoration controller UI would have options to upgrade and repair the decoration.

### Reproduction Steps

1. Create an instance of minecraft that has the minecolonies mod installed
2. Put blueprints from attached zip file into the minecraft instance's blueprints folder. [aygar.zip](https://github.com/ldtteam/minecolonies/files/9611983/aygar.zip)
3. Launch a new world with cheats on
4. Switch to creative mode
5. Acquire a Build tool.
6. Select the aygar style, in the scans section select the gatelarge1 blueprint and build the level 1 version.
9. Right-click on the decoration controller.
10. decoration controller UI shows no buttons and log states that it is unable to find the gatelarge1 file.

### Logs

https://gist.github.com/dupontct/fc7e7f1948c188eb6c162eab1b289873

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by BaronTilde** [comment]
I am on the MineColonies Official 1.19 server and am having similar problems.  I had originally placed a dozen Caledonia Alt roads and a Caledonia Big Field on my colony, and when wanting to finally upgrade found there is no 'upgrade' option, just build and repair.  Neither build or repair fix the issue, and sometimes build will cause the schematic to build at a different rotation from the controller.  

Trying to build a Lvl 2 Big Field results in the builder taking on a Lvl 1 Big Field job instead, on the server and in single player as well.
Placing the schematic in single player using the creative 'constructed' option will build the schematic at the requested 2-5 level.
![2022-09-23_09 29 03](https://user-images.githubusercontent.com/26805392/191993190-d3481d7b-c09c-4842-8364-1da830fc820b.png)

**Comment by MotionlessTrain** [comment]
> Trying to build a Lvl 2 Big Field results in the builder taking on a Lvl 1 Big Field job instead, on the server and in single player as well.
Placing the schematic in single player using the creative 'constructed' option will build the schematic at the requested 2-5 level.

That part is intended. The idea is that upgradable decorations, like buildings, get built at level 1 and upgraded to higher levels. Just like buildings, this is overridable with the "constructed" paste option

**Comment by MotionlessTrain** [comment]
From my tests, "Build Building" is actually the upgrade button. When I clicked it, it upgraded the field_big1 to field_big2

**Comment by MotionlessTrain** [comment]
Something seems to be completely broken regarding decoration controllers, or I'm missing something obvious:

I tried to scan a very simple decoration with decoration controller, and named it "test1". I modified it a bit and scanned it as "test2". They showed up in the build tool as upgradable decorations (name "test1", with level selection and the blueprints loaded correctly). I let a builder build the level 1 decoration. When the builder was done, the decoration controller had no data regarding a name at all, which causes it to not be upgradable.
I tried to explicitly set the controller as anchor block, and I tried moving both schematics in a different folder, both without effect
